### PR TITLE
Disallow system updates on closing windows

### PR DESCRIPTION
**Pull request by rueter37**
System changes are applied now after the window has been closed. This fixes a cosmetic bug in "Vampires Dawn" when you interact with the graves in the graveyard in "Asgars Schloss", the message window uses a different system graphic. But as soon as the window starts closing, the system graphic switches back to the default system graphic which looks a bit awkward. This PR fixes this by applying the change after the window has been closed. You can reproduce the bug and confirm the fix with this savegame: [Save01.zip](https://github.com/EasyRPG/Player/files/5939146/Save01.zip).

### CODE REVIEW

File: src/window_selectable.cpp [hunk 1]
```diff
@@ -123,8 +123,13 @@ void Window_Selectable::UpdateArrows() {
 }
 
 // Update
-void Window_Selectable::Update() {
-	Window_Base::Update();
+void Window_Selectable::Update(bool block_system_change) {
+	if (block_system_change) {
+		Window::Update();
+		Window_Base::UpdateMovement();
```
Comment (fmatthew5876) [inline src/window_selectable.cpp:129]: This is a gross hack, especially in #2451 where these 2 lines just appear and is totally not obvious why.

You should either:
1. Refactor the update routines so that all child classes must explicitly call an `UpdateSystemGraphic()` in their update routines.
2. Add some member variable flag or virtual function which says whether or not to update the system graphic and adjust it / call it as needed.

Basically lets make "should we update the system graphic?" a first class concept of the `Window` classes

**Review by fmatthew5876 — CHANGES_REQUESTED** [review]
See inline comment

**Comment by Ghabry** [comment]
> Refactor the update routines so that all child classes must explicitly call an UpdateSystemGraphic() in their update routines.

Imo just removing ``UpdateSystemGraphic`` would work. The only case where a live update is even possible is on map and battle.

The only way to update the system graphic is via ``CommandChangeSystemGraphics``.
Like e.g. with ```ChipsetUpdated`` there could be a ```SystemGraphicUpdated`` callback invoked on map and battle scene which invokes ``SetWindowskin(Cache::SystemOrBlack());`` on all windows.

**Comment by rueter37** [comment]
I'm sorry to disappoint but I'm giving up here. This is way above my skill level (Honestly I don't know why I have tried that at all in the first place).

**Comment by Ghabry** [comment]
Please don't take this personal. We appreciate all your contributions!

Even if they don't meet our "standards" they are usually a great base to work on (or give a hint where the problem is).

**Comment by rueter37** [comment]
Thank you for the positive response despite my endless negativity, I really appreciate it! Of course I will continue contributing to the EasyRPG project, because as a "Vampires Dawn" fan (and RPG Maker 2k/2k3 games in general - for nostalgia reasons) I have a genuine personal interest that EasyRPG reaches the 1.0 release someday. As an amateur in programming I try to stick to the issues which are easier to fix, but I have underestimated this one.
